src/main/java/com/example/gradecalculator/controller/SubjectController.java

```diff
@@ -0,0 +1,119 @@
+package com.example.gradecalculator.controller;
+
+import com.example.gradecalculator.entities.SchoolYear;
+import com.example.gradecalculator.entities.Subject;
+import com.example.gradecalculator.entities.User;
+import com.example.gradecalculator.entities.UserSubject;
+import com.example.gradecalculator.mapper.SubjectMapper;
+import com.example.gradecalculator.model.SubjectTO;
+import com.example.gradecalculator.repository.SchoolYearRepository;
+import com.example.gradecalculator.repository.SubjectRepository;
+import com.example.gradecalculator.repository.UserRepository;
+import com.example.gradecalculator.repository.UserSubjectRepository;
+import com.example.gradecalculator.service.SubjectService;
+import com.example.gradecalculator.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.*;
+
+@Controller
+public class SubjectController {
```

Comment:
please double check if all methods are needed and used in your workflow, so that no unnecessary code was published. also remove the correspoinding HTML file if it has no use